subsys/net/lib/download_client/src/download_client.c

```diff
@@ -247,7 +261,8 @@ static int client_connect(struct download_client *dl, const char *host,
 
 	if ((dl->proto == IPPROTO_TLS_1_2 || dl->proto == IPPROTO_DTLS_1_2)
 	     && (dl->config.sec_tag != -1)) {
-		err = socket_sectag_set(*fd, dl->config.sec_tag);
+		err = socket_sectag_set(*fd, dl->config.sec_tag,
+			dl->config.set_tls_hostname ? host : NULL);
```

Comment:
`host` may also contain scheme and port number, so those will have to be removed before setting the host name to the socket:

```
/**
 * @brief Establish a connection to the server.
 *
 * @param[in] client	Client instance.
 * @param[in] host	Name of the host to connect to, null-terminated.
 *			Can include scheme and port number, defaults to
 *			HTTP or HTTPS if no scheme is provided.
 * @param[in] config	Configuration options.
 *
 * @retval int Zero on success, a negative error code otherwise.
 */
int download_client_connect(struct download_client *client, const char *host,
			    const struct download_client_cfg *config);
```